### DISCUSSION TITLE
Error: Exec format error

### DISCUSSION THREAD

**Discussion by emeliada** [ipxe/ipxe#515] [Q&A]
I'm attempting to PXE boot from a newly configured FOG server and am getting Error code 2e0220. I checked the ipxe.org posting for this and have verified that I am on the newest version of iPXE. Any ideas
![ipxe](https://user-images.githubusercontent.com/94080710/141194842-2e4515fa-eaa3-47d6-aa20-b7f57bc86b1e.jpg)
?

**Comment by NiKiZe** [comment]
First, take a look in the script and try to figure out what it actually tries to do. Then follow the http://ipxe.org/2e022001 link that you see in the message and investigate what is wrong, maybe it is as simple as you are using an iPXE image without params command enabled?

**Reply by NiKiZe** [reply]
As mentioned on that page:
> * Ensure you've entered “kernel” before a line if you're executing a kernel.
> * Check the spelling of the command.
> * Check that the command exists in the list of iPXE commands.
> * Check that the command is enabled in your build of iPXE. Some commands are not enabled by default. For example, the vcreate command, which creates a VLAN interface, is available only when the build option VLAN_CMD is defined.

You are missing support for `params`, the script dies due to that command is missing, at the same time I guess that your iPXE has an embedded script that shows that chainload failed message.

If you are trying to use an existing solution than you should probably use that full solution with matching configurations.
If you want to try a more recent iPXE build then you should make sure that it is built to match the solution (in this case FOG)

**Comment by stappersg** [comment]
Tell more about the `default.ipxe`.
In case it is as a text file (a script) share the content with us.

Yes that is rephrase of
> First, take a look in the script and try to figure out what it actually does.

**Comment by emeliada** [comment]
Here is a screenshot of the default.ipxe file. The IP it has listed is not the same as the host i'm wondering if that could be part of the problem? 
![image](https://user-images.githubusercontent.com/94080710/142215484-bdcce87f-0e17-42f4-8125-bf28fa19d887.png)

**Reply by NiKiZe** [reply]
This file does not match the output that we see in the first image. (The chainloading failed message is missing)

Since you are using FOG and their script files you should probably also use their ipxe binaries. the reason for mentioning this is that you are using params in the script, but are using a iPXE binary that is build without params support.
